.github/workflows/ci.yml

```diff
@@ -0,0 +1,97 @@
+name: CI
+
+on:
+  push:
+    branches:
+    - main
+    tags:
+    - 'v*'
+  pull_request: {}
+
+jobs:
+  unit-tests:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: setup Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: '1.15'
+
+      - name: run Go unit tests
+        run: make test
+
+  e2e-tests:
+    needs: unit-tests
+    runs-on: ubuntu-latest
+    env:
+      WASM_BINARY_NAME: policy.wasm
+    steps:
+      #############################
+      # Preparing the environment
+      #############################
+      - uses: actions/checkout@v2
+
+      - name: Download policy-testdrive
+        uses: i3h/download-release-asset@v1
+        with:
+          owner: kubewarden
+          repo: policy-server
+          tag: v0.1.1
+          file: policy-testdrive-linux-amd64
+
+      - name: Setup BATS
+        uses: mig4/setup-bats@v1
+        with:
+          bats-version: 1.2.1
+
+      - name: Download wasm-to-oci
+        uses: i3h/download-release-asset@v1
+        with:
+          owner: engineerd
+          repo: wasm-to-oci
+          tag: v0.1.1
+          file: linux-amd64-wasm-to-oci
+      - name: fix wasm-to-oci permissions
+        run: |
+          chmod 755 linux-amd64-wasm-to-oci
+
+      #############################
+      # Building the Wasm module
+      #############################
+
+      # TODO: uncomment once tinygo 0.18.0 is released
+      #-
+      #  name: Install tinygo
+      #  run: |
+      #    wget https://github.com/tinygo-org/tinygo/releases/download/v0.17.0/tinygo_0.17.0_amd64.deb
+      #    sudo dpkg -i tinygo_0.17.0_amd64.deb
+      #-
+      #  name: Build Wasm module
+      #  run: |
+      #    tinygo build -o ingress-policy.wasm -target=wasi -no-debug .
+      - name: build policy
+        uses: docker://ghcr.io/kubewarden/tinygo/tinygo-dev:w14-2021
+        with:
+          args: tinygo build -o ${{ env.WASM_BINARY_NAME }} -target=wasi -no-debug .
+
+
+      #############################
+      # Test Wasm module
+      #############################
+      - name: run e2e tests
+        run: |
+          chmod 755 policy-testdrive-linux-amd64
+          ln -s policy-testdrive-linux-amd64 policy-testdrive
```

Comment:
I can see this as an external reusable GitHub action. Example: very soon we will replace `policy-testdrive` with `kwctl` to perform e2e testing in policies. Having a reusable action would allow us to only click the switch once on the action itself, without having to fix all policies.

Reply:
We will have to version the action and have to go through all policies anyway very probably, but you get the idea... is less code to replace :)

Reply:
Good point, I think I can handle that. Gimme some hours